### PR TITLE
Some structs refactor work

### PR DESCRIPTION
**Pull request by maqi**
This PR contains the work of:
1, rename SectionMembers to SectionPeers, MemberState to PeerState
2, rename quorum to majority
3, make Peer contains the age info

### CODE REVIEW

File: src/messages/mod.rs [hunk 1]
```diff
@@ -123,6 +124,7 @@ impl Message {
     /// Creates a signed message from single node.
     pub(crate) fn single_src(
         keypair: &Keypair,
+        age: u8,
```
Comment (dirvine) [inline src/messages/mod.rs:127]: Should this not take &Node instead of KeyPair + age?

File: src/messages/mod.rs [hunk 2]
```diff
@@ -136,6 +138,7 @@ impl Message {
         let signature = crypto::sign(&serialized, keypair);
         let src = SrcAuthority::Node {
             public_key: keypair.public,
+            age,
```
Comment (dirvine) [inline src/messages/mod.rs:141]: same here take &Peer instead of key + age ?

Comment (maqi) [reply src/messages/mod.rs:141]: Peer contains `socket addr` as well, which no need to be in `SrcAuthority::Node` ?

File: src/messages/mod.rs (hunk 1 above)
Comment (maqi) [reply src/messages/mod.rs:127]: keypair is from self, and `age` is fetched from SectionPeers info.
I thought Node won't contain the info of age ?

File: src/node/stage/bootstrapping.rs [hunk 3]
```diff
@@ -162,6 +162,7 @@ impl Bootstrapping {
 
         let message = Message::single_src(
             &self.node_info.keypair,
+            MIN_AGE,
```
Comment (madadam) [inline src/node/stage/bootstrapping.rs:165]: When we are relocating, this won't be `MIN_AGE`. You can fetch the correct age from `self.relocate_details`.

File: src/node/stage/joining.rs [hunk 4]
```diff
@@ -235,6 +235,7 @@ impl Joining {
         let variant = Variant::JoinRequest(Box::new(join_request));
         let message = Message::single_src(
             &self.node_info.keypair,
+            MIN_AGE,
```
Comment (madadam) [inline src/node/stage/joining.rs:238]: Same here

File: src/peer.rs [hunk 5]
```diff
@@ -15,14 +15,15 @@ use xor_name::XorName;
...
+    name: XorName,
+    addr: SocketAddr,
+    pub age: u8,
 }
 
 impl Peer {
-    /// Creates a new `Peer` given a `Name` and a `ConnectionInfo`.
-    pub fn new(name: XorName, addr: SocketAddr) -> Self {
-        Self { name, addr }
+    /// Creates a new `Peer` given `Name`, `ConnectionInfo` and `age`.
```
Comment (madadam) [inline src/peer.rs:24]: ConnectionInfo -> SocketAddr

File: src/node/stage/bootstrapping.rs [hunk 6]
```diff
@@ -160,9 +160,15 @@ impl Bootstrapping {
             None => self.node_info.name(),
         };
 
+        let age = if let Some(signed_relocation_details) = &self.relocate_details {
+            signed_relocation_details.relocate_details().age
+        } else {
+            MIN_AGE
+        };
```
Comment (madadam) [inline src/node/stage/bootstrapping.rs:167]: You could have done it together with the previous match:
```rust
let (destination, age) = match &self.relocate_details {
    Some(details) => (*details.destination(), details.relocate_details().age),
    None => (self.node_info.name(), MIN_AGE)
};
```

File: src/peer.rs [hunk 7]
```diff
@@ -36,6 +36,16 @@ impl Peer {
...
 
+    /// Returns the age.
+    pub fn age(&self) -> u8 {
+        self.age
+    }
+
+    // Converts this info into one with the input age.
+    pub fn into_age(self, age: u8) -> Self {
+        Self { age, ..self }
+    }
```
Comment (madadam) [inline src/peer.rs:47]: The `into_*` convention usually signifies that you are converting one type into another one. Here you are only changing the age which makes it a confusing name.  How about `change_age` or `with_age` ?

**Review by dirvine — APPROVED** [review]

**Review by madadam — APPROVED** [review]

**Pull request merged**